### PR TITLE
replace golint with revive

### PR DESCRIPTION
**Pull request by zirain**
replace golint with revive

part of https://github.com/istio/istio/issues/38201

need merge https://github.com/istio/istio/pull/38287 and https://github.com/istio/release-builder/pull/1023 first.

### CODE REVIEW

File: files/common/config/.golangci.yml [hunk 1]
```diff
@@ -70,6 +70,67 @@ linters-settings:
   golint:
```
Comment (ericvn) [inline files/common/config/.golangci.yml:70]: I think we can remove the `golint` section.

File: files/common/config/.golangci.yml [hunk 2]
```diff
@@ -70,6 +70,67 @@ linters-settings:
...
     # minimal confidence for issues, default is 0.8
     min-confidence: 0.0
+  revive:
```
Comment (ericvn) [inline files/common/config/.golangci.yml:73]: In my earlier testing I had just used the same two lines for `revive` but I do like this calling out better.

Also, can we move this to the correct place alphabetically.

**Comment by ericvn** [comment]
Add DNM until the prerequisites PRs are merged.

File: files/common/config/.golangci.yml (hunk 2 above)
Comment (zirain) [reply files/common/config/.golangci.yml:73]: will fix the place later, also wait for John's agreement on istio repo changes.

File: files/common/config/.golangci.yml (hunk 1 above)
Comment (zirain) [reply files/common/config/.golangci.yml:70]: ok, will remove later.

**Comment by zirain** [comment]
@ericvn I think it's ok to get this merged.

**Comment by ericvn** [comment]
It looks like there are other lint issues in a few of the other repos as well. I'll work on some updates.

**Pull request merged**